query.go

```diff
@@ -673,15 +680,68 @@ func (q *Query) SetBufferVar(attribute string, offset []uint64,
 		q.tiledbQuery,
 		cAttribute,
 		(*C.uint64_t)(coffset),
-		&coffsetSize,
+		(*C.uint64_t)(unsafe.Pointer(&offsetSize)),
 		cbuffer,
 		(*C.uint64_t)(unsafe.Pointer(&bufferSize)))
 
 	if ret != C.TILEDB_OK {
-		return nil, fmt.Errorf("Error setting query var buffer: %s",
+		return nil, nil, fmt.Errorf("Error setting query var buffer: %s",
 			q.context.LastError())
 	}
-	return &bufferSize, nil
+
+	q.resultBufferElements[attribute] =
+		[2]*uint64{&offsetSize, &bufferSize}
+
+	return &offsetSize, &bufferSize, nil
+}
+
+// ResultBufferElements returns the number of elements in the result buffers
+// from a read query.
+// This is a map from the attribute name to a pair of values.
+// The first is number of elements (offsets) for var size attributes, and the
+// second is number of elements in the data buffer. For fixed sized attributes
+// (and coordinates), the first is always 0.
+func (q *Query) ResultBufferElements() (map[string][2]uint64, error) {
+	elements := make(map[string][2]uint64, 0)
+
+	// Will need the schema to infer data type size for attributes
+	schema, _ := q.array.Schema()
```

Comment:
We should check for error and return if we error on getting the schema

Reply:
checked